File: src/types/EventRequest.ts
// src/types/EventRequest.ts
/** Статус заявки */
export type RequestStatus = 'PENDING' | 'APPROVED' | 'REJECTED';
/** Тип заявки: новое событие или правка существующего */
export type RequestType   = 'CREATE' | 'EDIT';

/** DTO, которое backend отдаёт на /event-requests/** */
export interface EventRequestDto {
  id:               number;
  title:            string;
  description:      string;
  date:             string | null;

  category:         string | null;
  format:           string | null;   // "online" | "offline"
  address:          string | null;
  city:             string | null;
  latitude:         number | null;
  longitude:        number | null;
  conferenceLink:   string | null;
  capacity:         number | null;
  level:            string | null;

  type:             RequestType;     // 'CREATE' | 'EDIT'
  status:           RequestStatus;   // 'PENDING' | 'APPROVED' | 'REJECTED'
  authorUsername:   string | null;
  originalEventId?: number | null;   // для EDIT
}

File: events-backend/src/main/java/com/example/eventsbackend/model/EventRequest.java
package com.example.eventsbackend.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Data
@Entity
@Table(name = "event_requests")
public class EventRequest {
    @Id @GeneratedValue
    private Long id;
    private String title;
    private String description;
    private LocalDateTime date;
    private String category;
    private String format;          // "online"/"offline"
    private String address;
    private String city;
    private Double latitude;
    private Double longitude;
    private String conferenceLink;
    private Integer capacity;
    private String level;

    @OneToMany(mappedBy = "request",
            cascade = CascadeType.ALL,
            orphanRemoval = true)
    @JsonIgnoreProperties("request")
    private List<EventRequestImage> images = new ArrayList<>();

    @Enumerated(EnumType.STRING)
    private RequestType type;       // CREATE | EDIT
    @Enumerated(EnumType.STRING)
    private RequestStatus status;   // PENDING | APPROVED | REJECTED
    @ManyToOne
    private User author;
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "original_event_id")
    private Event originalEvent; // nullable для CREATE
}
